File: frontend/src/api/ai.ts
import { api } from "./client";
import type { PriorityLevel, QuickReplySuggestion } from "../types";

export type AISource = "ai" | "heuristic";

export interface SmartTagRequest {
  subject: string;
  body_text: string;
  sender_address: string;
}

export interface PrioritizeRequest extends SmartTagRequest {
  is_reply?: boolean;
}

export interface QuickReplyRequest {
  subject: string;
  body_text: string;
  sender_name?: string | null;
  tone?: "professional" | "friendly" | "brief";
}

export interface AIStatus {
  enabled: boolean;
  provider: string;
  model: string | null;
}

export const aiApi = {
  status: () => api.get<AIStatus>("/ai/status"),
  tag: (payload: SmartTagRequest) =>
    api.post<{ tags: string[]; confidence: number | null; source: AISource }>("/ai/tag", payload),
  prioritize: (payload: PrioritizeRequest) =>
    api.post<{ priority: PriorityLevel; score: number; reasons: string[]; source: AISource }>(
      "/ai/prioritize",
      payload,
    ),
  quickReply: (payload: QuickReplyRequest) =>
    api.post<{ suggestions: QuickReplySuggestion[]; source: AISource }>("/ai/quick-reply", payload),
};

File: backend/app/schemas/ai.py
from pydantic import BaseModel, Field

AISource = str  # "ai" | "heuristic"


class SmartTagRequest(BaseModel):
    subject: str
    body_text: str
    sender_address: str


class SmartTagResponse(BaseModel):
    tags: list[str]
    confidence: float | None = None
    source: AISource = "heuristic"


class PrioritizeRequest(BaseModel):
    subject: str
    body_text: str
    sender_address: str
    is_reply: bool = False


class PrioritizeResponse(BaseModel):
    priority: str  # "low" | "normal" | "high" | "urgent"
    score: float
    reasons: list[str] = Field(default_factory=list)
    source: AISource = "heuristic"


class QuickReplyRequest(BaseModel):
    subject: str
    body_text: str
    sender_name: str | None = None
    tone: str = "professional"  # "professional" | "friendly" | "brief"


class QuickReplySuggestion(BaseModel):
    label: str
    body: str


class QuickReplyResponse(BaseModel):
    suggestions: list[QuickReplySuggestion]
    source: AISource = "heuristic"


class AIStatusResponse(BaseModel):
    enabled: bool
    provider: str
    model: str | None = None
